client/app/components/account_config_mailboxes.coffee

```diff
@@ -158,57 +76,98 @@ module.exports = AccountConfigMailboxes = React.createClass
         else if Object.keys(@props.errors).length isnt 0
             div className: 'alert alert-warning', t 'account errors'
 
-
-    renderMailboxChoice: (labelText, box) ->
-        if @props.id? and @props.mailboxes.value isnt ''
-            errorClass = if @props[box].value? then '' else 'has-error'
-            div className: "form-group #{box} #{errorClass}",
-                label
-                    className: 'col-sm-2 col-sm-offset-2 control-label',
-                    labelText
-                div className: 'col-sm-3',
-                    MailboxList
-                        allowUndefined: true
-                        mailboxes: @state.mailboxesFlat
-                        selectedMailboxID: @props[box].value
-                        onChange: (mailbox) => @onMailboxChange mailbox, box
-
+    renderTableHeader: ->
+        li className: 'row box title', key: 'title',
+            span className: "col-xs-1", ''
+            span className: "col-xs-1", ''
+            span className: "col-xs-6", ''
+            span className: "col-xs-1", ''
+            span className: "col-xs-1 text-center",
+                t 'mailbox title total'
+            span className: "col-xs-1 text-center",
+                t 'mailbox title unread'
+            span className: "col-xs-1 text-center",
+                t 'mailbox title new'
+
+    renderTableFooter: ->
+        li className: "row box new", key: 'new',
+
+            span
+                className: "col-xs-1 box-action add"
+                onClick: @addMailboxClicked
+                title: t("mailbox title add"),
+                    i className: 'fa fa-plus'
+
+            span
+                className: "col-xs-1 box-action cancel"
+                onClick: @resetMailboxClicked
+                title: t("mailbox title add cancel"),
+                    i className: 'fa fa-undo'
+
+            div className: 'col-xs-6',
+                input
+                    id: 'newmailbox',
+                    type: 'text',
+                    className: 'form-control',
+                    placeholder: t "account newmailbox placeholder"
+                    valueLink: @linkState 'newMailboxName'
+                    onKeyDown: @onNewMailboxKeyDown
+
+            label
+                className: 'col-xs-2 text-center control-label',
+                t "account newmailbox parent"
+
+            div className: 'col-xs-2 text-center',
+                MailboxPicker
+                    allowUndefined: true
+                    valueLink: @linkState 'newMailboxParent'
+                    mailboxes: @props.editedAccount.get('mailboxes')
+
+    renderMailboxChoice: (which) ->
+        property = "#{which}Mailbox"
+        labelText = t "account #{which} mailbox"
+
+        className = "form-group #{which}"
+        className += 'has-error' unless @props.editedAccount.get property
```

Comment:
@aenario is it normal to have `#{which}` and `has-error` with no space to separate them?